autogluon/task/text_classification/dataset.py

```diff
@@ -116,31 +120,168 @@ def get_dataset(self, segment='train'):
+        Evaluation metrics of the task.
+    is_pair : bool
+        Whether the task deals with sentence pairs or single sentences.
+    label_alias : dict
+        label alias dict, some different labels in dataset actually means
+        the same. e.g.: {'contradictory':'contradiction'} means contradictory
+        and contradiction label means the same in dataset, they will get
+        the same class id.
+    """
+    def __init__(self, class_labels, metrics, is_pair, label_alias=None):
+        self.class_labels = class_labels
+        self.metrics = metrics
+        self.is_pair = is_pair
+        self.label_alias = label_alias
+
+    def get_dataset(self, segment='train'):
+        """Get the corresponding dataset for the task.
+
+        Parameters
+        ----------
+        segment : str, default 'train'
+            Dataset segments.
+
+        Returns
+        -------
+        Dataset : the dataset of target segment.
+        """
+        raise NotImplementedError
 
     def dataset_train(self):
-        return 'train', self.trainset
+        """Get the training segment of the dataset for the task.
+
+        Returns
+        -------
+        tuple of str, Dataset : the segment name, and the dataset.
+        """
+        return 'train', self.get_dataset(segment='train')
 
     def dataset_dev(self):
-        return 'dev', self.valset
+        """Get the development (i.e. validation) segment of the dataset for this task.
+
+        Returns
+        -------
+        tuple of str, Dataset : the segment name, and the dataset.
+        """
+        return 'dev', self.get_dataset(segment='dev')
+
+    def dataset_test(self):
+        """Get the test segment of the dataset for the task.
+
+        Returns
+        -------
+        tuple of str, Dataset : the segment name, and the dataset.
+        """
+        return 'test', self.get_dataset(segment='test')
+
+
+class CustomTSVClassificationTask(AbstractCustomTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
+        Any valid string path is acceptable.
+    sep : str
+        Delimiter to use.
+    header : int, list of int, default 'infer'
+        Row number(s) to use as the column names, and the start of the
+        data.
+    names : array-like, optional
+        List of column names to use.
+    usecols : list-like or callable, optional
+        Return a subset of the columns.
+    For other keyword arguments, please see
+        <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self._label_column_id = self._index_label_column(**kwargs)
+        self.class_labels = list(set([sample[self._label_column_id] for sample in self.dataset]))
+        self.metric = Accuracy()
+        super(CustomTSVClassificationTask, self).__init__(self.class_labels, self.metric, self.is_pair)
```

Comment:
There are at least two reasons: 1) it provides better semantic of the code. By specifying self.pair = False then pass it to parent class, it gives user the better understanding of what this argument is rather than a random variable False; 2) it provides better flexibility. The user could modify it based their usage.

Reply:
Same reply. There are at least two reasons: 1) it provides better semantic of the code. By specifying self.class_labels then pass it to parent class, it gives user the better understanding of what this argument is rather than a random variable False; 2) it provides better flexibility. The user could modify it based their usage. 3) It also provides better code style, obviously, 
__init__(self.class_labels, self.metric, self.is_pair) is better regarding style compared to 
__init__(
list(set([sample[self._label_column_id] for sample in self.dataset]))
, self.metric, self.is_pair)

Reply:
In the future, we would have more custom text format supported in the nlp module. 
AbstractCustomTask is the abstract class that every new dataset support inheriting from.
For example, now the tsv/csv dataset supports inherit from the abstract class.
As replied in the previous responses, the metric can be a composed metric including multiple metrics.
The current tsv dataset supports accuracy for now.

Reply:
I don't understand the points you are making here. This is Python 101 to avoid repeatedly declaring variables. That is one of the major points of using class inheritance.

Reply:
We will have to discuss this in a meeting. It appears there is some deeper misunderstanding going on.